src/start/RealmScreen.js

```diff
@@ -24,7 +24,7 @@ type Props = $ReadOnly<{|
   // `navigation` prop for free, with the stack-nav shape.
   navigation: NavigationStackProp<{|
     ...NavigationStateRoute,
-    params?: {|
```

Comment:
Before a770a348d, there was an easily overlooked case where `params`
    would be missing: although we included `params` whenever we
    dispatched an action to navigate to the 'realm' route, there was
    still a case in `navReducer` in which the nav state was constructed
    with the 'realm' route active but with no params.

I think this isn't actually true -- though the code was certainly tricky. We had `return getStateForRoute('realm');`... but then that function's definition had this:
```js
 /**
  * Get the initial state for the given route.
  *
  * Private; exported only for tests.
  */
 export const getStateForRoute = (route: string): NavigationState => {
   // TODO: this is kind of a hack!  Refactor to a better way.
   //  * Perhaps pass `initial: true` unconditionally in this initialization
   //    code, to all routes?  Then that'd just be part of the interface of
   //    making a screen work as an initial screen.
   //  * Alternatively, we could replace the whole system of `ModalNavBar`,
   //    `canGoBack`, etc., with our own "navigation view" that would be more
   //    directly integrated into the navigation framework:
   //      https://reactnavigation.org/docs/en/2.x/navigation-views.html
   const params = route === 'realm' ? { initial: true } : undefined;
```

So in fact when we went to the 'realm' route via the logic in `navReducer`, we'd still pass a params object.

(Hmm, and I have the feeling that we've had this conversation before, sometime in the last few weeks. Am I imagining that? The lesson is probably that upon discussing it then, it would have been smart to go promptly edit the code to make it clearer -- with a comment over on the RealmScreen side, if nothing else, but better by clarifying the actual code.)

Reply:
(Also the type of that `params` object isn't quite right: comparing `navigateToRealmScreen` and this `navReducer` hack, it should be `{| realm?: URL | void, initial: boolean | void |}`.)

Reply:
> (Hmm, and I have the feeling that we've had this conversation before, sometime in the last few weeks. Am I imagining that? The lesson is probably that upon discussing it then, it would have been smart to go promptly edit the code to make it clearer -- with a comment over on the RealmScreen side, if nothing else, but better by clarifying the actual code.)

Ah, right—that conversation was over at https://github.com/zulip/zulip-mobile/pull/4273#discussion_r499114689. It looks like (even in a34d3ba16) I skipped right over your observation in that discussion that `params` would always be present, oops. I have a vague memory that I had actually encountered an error while testing the branch, which prompted me to remove https://github.com/zulip/zulip-mobile/commit/6f6a3ee4a3378d49dba6305a58b2c3a78304a40b from it (and comment at https://github.com/zulip/zulip-mobile/pull/4273#discussion_r499095804), and that might have led me to skip over that part. I guess, if there was an error, it was about something else, or else the `navReducer` code just jumped out at me as a potential reason for `params` to be missing, even if that turned out to be wrong.

Reply:
> (Also the type of that `params` object isn't quite right: comparing `navigateToRealmScreen` and this `navReducer` hack, it should be `{| realm?: URL | void, initial: boolean | void |}`.)

Hmm, now that `navReducer` is gone and we're using `navigateToRealmScreen` and `resetToRealmScreen`, I think it should be `{| realm: URL | void, initial: boolean | void |}`, maybe? In both those functions, if the caller doesn't pass `realm` or `initial`, it ends up being present in the action as `undefined`.